Sming/Arch/Esp8266/Components/esp_no_wifi/user_interface.c

```diff
@@ -0,0 +1,540 @@
+#include <esp_systemapi.h>
+#include <eagle_soc.h>
+
+bool protect_flag;
+bool timer2_ms_flag;
+uint32_t system_param_sector_start;
+uint32_t WdevTimOffSet;
+static bool os_print_enabled = true;
+
+void ets_update_cpu_frequency(uint8_t freq);
+
+uint8_t system_upgrade_userbin_check(void)
+{
+	return 0x00;
+}
+
+uint32 system_get_time(void)
+{
+#define WPS_TIME_REG 0x3ff20c00
+	return READ_PERI_REG(WPS_TIME_REG) + WdevTimOffSet;
+}
+
+bool system_rtc_mem_write(uint8 des_addr, const void* src_addr, uint16 save_size)
+{
+	return false;
+}
+
+/*
+Disassembly of section .text.system_rtc_mem_write:
+
+00000000 <system_rtc_mem_write-0x4>:
+   0:	00 0e 00 60
+
+00000004 <system_rtc_mem_write>:
+   4:	bfa052               	movi	a5, 191
+   7:	143527               	bltu	a5, a2, 1f <system_rtc_mem_write+0x1b>
+   a:	139c                	beqz.n	a3, 1f <system_rtc_mem_write+0x1b>
+   c:	370c                	movi.n	a7, 3
+   e:	146030               	extui	a6, a3, 0, 2
+  11:	a6cc                	bnez.n	a6, 1f <system_rtc_mem_write+0x1b>
+  13:	1192e0               	slli	a9, a2, 2
+  16:	00a382               	movi	a8, 0x300
+  19:	c08890               	sub	a8, a8, a9
+  1c:	03a847               	bge	a8, a4, 23 <system_rtc_mem_write+0x1f>
+  1f:	020c                	movi.n	a2, 0
+  21:	f00d                	ret.n
+  23:	0a0747               	bnone	a7, a4, 31 <system_rtc_mem_write+0x2d>
+  26:	450c                	movi.n	a5, 4
+  28:	414240               	srli	a4, a4, 2
+  2b:	a04450               	addx4	a4, a4, a5
+  2e:	f44040               	extui	a4, a4, 0, 16
+  31:	418240               	srli	a8, a4, 2
+  34:	e89c                	beqz.n	a8, 56 <system_rtc_mem_write+0x52>
+  36:	fff271               	l32r	a7, 0 <system_rtc_mem_write-0x4>
+  39:	040c                	movi.n	a4, 0
+  3b:	624a                	add.n	a6, a2, a4
+  3d:	a05430               	addx4	a5, a4, a3
+  40:	a06670               	addx4	a6, a6, a7
+  43:	0020c0               	memw
+  46:	0558                	l32i.n	a5, a5, 0
+  48:	0020c0               	memw
+  4b:	c06652               	s32i	a5, a6, 0x300
+  4e:	441b                	addi.n	a4, a4, 1
+  50:	744040               	extui	a4, a4, 0, 8
+  53:	e43487               	bltu	a4, a8, 3b <system_rtc_mem_write+0x37>
+  56:	120c                	movi.n	a2, 1
+  58:	f00d                	ret.n
+*/
```

Comment:
This is an example where I would like to just be able to use the implementation from libmain, but it's not in a separate section. This disassembly clearly shows there are no relocations or external dependencies so in theory I should be able to pull this symbol out as a separate object file. Any clues as to how I can do this?

Reply:
NB. I've added the `sdk-disassemble` build target so the full disassembly can be generated. The above example will be in `sdk/main/user_interface.asm`.